torch/functional.py

```diff
@@ -71,6 +71,49 @@ def broadcast_tensors(*tensors):
     return _VF.broadcast_tensors(tensors)  # type: ignore
 
 
+def broadcast_shape(*shapes):
+    r"""
+
+    Similar to :func:`broadcast_tensors` but for shapes.
+
+    This is roughly equivalent to
+    `torch.broadcast_tensors(*map(torch.empty, shapes)).shape`
+    but avoids the need create to intermediate tensors. This is useful for
+    broadcasting tensors of common batch shape but different rightmost shape,
+    e.g. to broadcast mean vectors with covariance matrices.
+
+    Example::
+
+        >>> mean = torch.randn(30, 3)
```

Comment:
I think you want a simpler example here. Without slicing and adding back more dimensions. Just to show what the function actually computes.

Reply:
Sure, I've simplified the example a bit. I do think slicing is important, since the main use case is in broadcasting with differing event shapes (non-batched shapes); otherwise you could simply use `broadcast_tensors()`. Let me know if you have suggestions for a yet-simpler motivating example. Maybe we can add two examples.